akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.io.StdIn
+import scala.util.{ Failure, Success }
+
+/**
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+trait HttpApp {
+
+  def main(args: Array[String]) = {
+    val config = createConfig()
+    implicit val system = createActorSystem(config)
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = system.dispatcher
+    postStartActorSystem(system)
+
+    val host = config.getString("akka.http.server.host")
+    val port = config.getInt("akka.http.server.port")
+
+    val bindingFuture = Http().bindAndHandle(
+      handler = route,
+      interface = host,
+      port = port,
+      settings = ServerSettings(config))
+
+    bindingFuture.onComplete {
+      case Success(_) ⇒
+        println(s"Server online at http://$host:$port/")
```

Comment:
These two should probably be moved into hooks so they can be overriden; someone somewhere is going to want a different error logger (more verbose? logback?) or difference success banner